module/icp/algs/aes/aes_impl.c

```diff
@@ -1584,34 +1584,21 @@ intel_aes_instructions_present(void)
 	unsigned eax, ebx, ecx, edx;
 	unsigned func, subfunc;
 
-	if (cached_result == -1) { /* first time */
-		/* check for an intel cpu */
-		func = 0;
-		subfunc = 0;
-
-		__asm__ __volatile__(
-		    "cpuid"
-		    : "=a" (eax), "=b" (ebx), "=c" (ecx), "=d" (edx)
-		    : "a"(func), "c"(subfunc));
-
-		if (memcmp((char *)(&ebx), "Genu", 4) == 0 &&
-		    memcmp((char *)(&edx), "ineI", 4) == 0 &&
-		    memcmp((char *)(&ecx), "ntel", 4) == 0) {
-			func = 1;
-			subfunc = 0;
-
-			/* check for aes-ni instruction set */
-			__asm__ __volatile__(
-			    "cpuid"
-			    : "=a" (eax), "=b" (ebx), "=c" (ecx), "=d" (edx)
-			    : "a"(func), "c"(subfunc));
-
-			cached_result = !!(ecx & INTEL_AESNI_FLAG);
-		} else {
-			cached_result = 0;
-		}
+	if (cached_result != -1) {
```

Comment:
(Nitpick) Remove the brackets

Reply:
don't check for Intel CPU. this would affect virtualised instances.